src/groupByKey.ts

```diff
@@ -2,6 +2,23 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T, K extends keyof T>(
+  items: T[],
+  key: K,
+): GroupsMap<T> {
+  const grouped: GroupsMap<T> = {};
+
+  items.map((item) => {
```

Comment:
will be better use `.reduce()` in this case without redundant variable `grouped`